### ISSUE TITLE
Publish Minor Version Tags to Docker

### ISSUE THREAD

**Issue by cperrin88** [prometheus/prometheus#13540]
### Proposal

At the moment you can only reference the docker images by their complete version, main or latest. It would be useful if you could also publish tags for the minor versions and maybe a major and an LTS tag. It would be useful fpr people who use tools like watchtower to be able to automatically update to the newes patch version.

Example:
For the tag `v2.43.3` you also publish a tag `v2.43` that points to the latest 2.43.yy release and maybe a `v2` that point to the newest 2.xx.yy release.

**Comment by darshanime** [comment]
Generally, prometheus stays away from "config management"-y things. Can you share about any major projects that do this though?

**Comment by cperrin88** [comment]
Pretty much every "official" image that has at least a SemVer like schema uses this pattern. See for exampe [MySQL](https://hub.docker.com/_/mysql) or [python](https://hub.docker.com/_/python)

**Comment by darshanime** [comment]
Gotcha. @roidelapluie @LeviHarrison what do you think about this? 

If we were to do this, we'll have to find out how to mark the wide variety of tags we have currently:

```
dev
0.20.0
v2.49.1
0.9.0rc1
checkout
0.13.0rc2
v0.35.0-rc1
v2.0.0-rc.3
v2.35.0-rc0
v2.0.0-beta.3
all-tests-pass
v2.0.0-alpha.0
v2.32.0-beta.0
v1.99.0-retract
v2.43.1+stringlabels
v2.43.0-rc.1+stringlabels
discovery-handle-discoverer-updates
```

**Comment by cperrin88** [comment]
It would already be useful if you only published a minor and mybe a major version docker tag with your production versions. You already have some logic for you latest tag here https://github.com/prometheus/promci/blob/main/actions/publish_release_images/action.yml#L48. Maybe you could hook it there.

**Comment by LeviHarrison** [comment]
Yeah, I think it makes sense to at least publish a separate major tag (eg `v2.50`) and an LTS tag.

**Comment by krajorama** [comment]
Hello from the bug scrub: I have a PR that implements the algorithm in bash, it's under discussion: https://github.com/prometheus/prometheus/pull/16324 .

**Comment by krajorama** [comment]
> Hello from the bug scrub: I have a PR that implements the algorithm in bash, it's under discussion: [#16324](https://github.com/prometheus/prometheus/pull/16324) .

closed in favor of simpler solution
